action/protocol/staking/handler_stake_migrate.go

```diff
@@ -0,0 +1,181 @@
+package staking
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	accountutil "github.com/iotexproject/iotex-core/action/protocol/account/util"
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/state"
+)
+
+const (
+	executionProtocolID = "smart_contract"
+)
+
+type (
+	executionProtocol interface {
```

Comment:
no difference from `protocol.ActionHandler`

Reply:
changed to call `HandleCrossProtocol`, which will reset caller nonce, avoiding nonce set error